### ISSUE TITLE
Is there any relation to the CausalInference.jl package?

### ISSUE THREAD

**Issue by sbacelar** [JuliaDynamics/CausalityTools.jl#285]

**Comment by kahaaga** [comment]
Hi @sbacelar! 

The CausalInference.jl package is related, but provides a much more limited and specialized set of functions. Here are some differences:

## CausalInference.jl provides:

- Top-level functions for the PC and FCI causal graph inference algorithms. 
- Tools for reasoning about directed acyclic graphs (which are the output of the PC and FCI algorithm)
- Three association measures: (partial) correlation, (conditional) mutual information (but only one estimator of the latter)
- Two independence tests: a gaussian one (which CausalityTools will also have when #249 is implemented), and one based on conditional mutual information.


## CausalityTools.provides

- Top level function for the optimal causation entropy (OCE) graph inference algorithm. Other graph algorithms on the way are the PCMCI (Runge et al) and the PC algorithm (the same as for CausalInference.jl)
- Currently, no tools for reasoning about directed acyclic graphs, but there are plans to implement some machinery similar to that of CausalInference.jl, or re-use it here.

The major difference to CausalInference.jl is how this package is strucured. It provides three conceptually different layers:
- Association measures. While CausalInference.jl only has two measures (CMI and correlation), we offer many more association measures. See [here](https://juliadynamics.github.io/CausalityTools.jl/dev/measures/) for an overview. 
- Many of our measures can be estimated in a plethora of different ways. For example, here you can compute conditional mutual information (`CMIShannon`) using about 15 different estimators from the literature (see overview [here](https://juliadynamics.github.io/CausalityTools.jl/dev/api/api_condmutualinfo/#Conditional-mutual-information-API)), both discrete and continuous and mixed. Information-based measures in this package tries to be compatible with the discrete estimators from [ComplexityMeasures.jl](https://github.com/JuliaDynamics/ComplexityMeasures.jl).
- Independence tests. See overview [here](https://juliadynamics.github.io/CausalityTools.jl/dev/independence/). Notably, `SurrogateTest` can be used with *any* association measure from the list above and *any* surrogate type from [TimeseriesSurrogates.jl](https://juliadynamics.github.io/TimeseriesSurrogates.jl/stable/), and `LocalPermutationTest` can be used with *any* conditional association measure from above.  In contrast, with CausalInference.jl, you only have two options: a Gaussian independence test using correlation, or a CMI-test with one (1) alternative for estimation. 
- Any `SurrogateTest` or `LocalPermutationTest` can be used as input to the OCE causal graph algorithm. Future implementations of the PC algorithm and related causal graph algorithms will also be equally flexible.
- Our tests are reproducible, because you can provide your own RNG instance to the independence tests.

I haven't counted, but with all the available options, there must be hundreds of ways of inferring causal graphs at the top-level already, using only the `OCE` algorithm. Some of these estimation routes have been explored in the literature, while most of them have not. But here you can pick and choose from any estimator of any measure from the literature (provided that it is implemented), and just apply it to whatever independence test or causal graph algorithm you want (provided the test or graph algorithm is implemented). 

In summary, the design philosophy of this package is *modularity* and reproducibility. It is a practical attempt at unifying all the various ways of computing associations between any kind of data, and determining if such associations are significant.

CausalInference.jl provides a very small subset of this functionality, with some very nice additional tools to reason about the resulting graphs. I suspect there will be almost complete overlap between the packages in the not-so-distant future. However, there is no way of easily merging the package, because of our modular API design. 

I think that should cover it! Feel free to drop any questions if there's something I missed.
